boards/arm/adafruit_feather_nrf52840_sense/doc/index.rst

```diff
@@ -0,0 +1,156 @@
+- SWD debug pads on bottom of PCB
+
+Supported Features
+==================
+
+The Adafruit Feather nRF52840 Sense board configuration supports the
+following hardware features:
+
++-----------+------------+----------------------+
+| Interface | Controller | Driver/Component     |
++===========+============+======================+
+| ADC       | on-chip    | adc                  |
++-----------+------------+----------------------+
+| CLOCK     | on-chip    | clock_control        |
++-----------+------------+----------------------+
+| FLASH     | on-chip    | flash                |
++-----------+------------+----------------------+
+| GPIO      | on-chip    | gpio                 |
++-----------+------------+----------------------+
+| I2C       | on-chip    | i2c                  |
++-----------+------------+----------------------+
+| MPU       | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| NVIC      | on-chip    | arch/arm             |
++-----------+------------+----------------------+
+| PWM       | on-chip    | pwm                  |
++-----------+------------+----------------------+
+| RADIO     | on-chip    | Bluetooth,           |
+|           |            | ieee802154           |
++-----------+------------+----------------------+
+| RTC       | on-chip    | system clock         |
++-----------+------------+----------------------+
+| SPI       | on-chip    | spi                  |
++-----------+------------+----------------------+
+| UART      | on-chip    | serial               |
++-----------+------------+----------------------+
+| USB       | on-chip    | usb                  |
++-----------+------------+----------------------+
+| WDT       | on-chip    | watchdog             |
++-----------+------------+----------------------+
+
+Other hardware features have not been enabled yet for this board.
+
+Connections and IOs
+===================
+
+The `Adafruit Feather nRF52840 Sense Learn site`_ has detailed
+information about the board including `pinouts`_ and the `schematic`_.
+
+LED
+---
+
+* LED0 (red) = P1.9
+* LED1 (blue) = P1.10
+
+Push buttons
+------------
+
+* SWITCH = P1.02
+* RESET = P0.18
+
+Programming and Debugging
+*************************
+
+The Feather Sense ships with a BOSSA compatible UF2 bootloader.  The
+bootloader can be entered by quickly tapping the reset button twice.
+
+Additionally, if :code:`CONFIG_USB_CDC_ACM` is enabled then the bootloader
+will be entered automatically when you run :code:`west flash`.
+
+Flashing
+========
+
+#. Build the Zephyr kernel and the :zephyr:code-sample:`blinky` sample application.
+
+   .. zephyr-app-commands::
+      :zephyr-app: samples/basic/blinky
+      :board: adafruit_feather_nrf52840_sense
+      :goals: build
+      :gen-args: -DCONFIG_USB_CDC_ACM=y
+      :compact:
+
+#. Connect the board to your host computer using USB
+
+#. Tap the reset button twice quickly to enter bootloader mode
+
+#. Flash the image.
```

Comment:
`west build` succeeds, but `west flash` fails with
```
-- west flash: using runner bossac
No device found on /dev/ttyACM0
FATAL ERROR: command exited with status 1: /home/ben/zephyr-sdk-0.16.3/sysroots/x86_64-pokysdk-linux/usr/bin/bossac -p /dev/ttyACM0 -R -e -w -v -b /home/ben/zephyrproject/zephyr/build/zephyr/zephyr.bin -o 49152
```

I'm not familiar with BOSSA; is it not capable of flashing without a serial adapter in addition to the USB cable? I note that the Adafruit ItsyBitsy docs describe connecting both the USB cable and a serial adapter.

Reply:
I've gotten closest to success building a UF2 file and copying that to the bootloader mode mass storage device: the mass storage device unmounts as normal, but the board is not responsive

```
$ west build -p always -b adafruit_feather_nrf52840_sense samples/hello_world -- -DCONFIG_BUILD_OUTPUT_UF2=y -DCONFIG_USB_CDC_ACM=y
$ cp build/zephyr/zephyr.uf2 /media/ben/FTHRSNSBOOT
$ arduino-cli monitor -p /dev/ttyACM0 
Port monitor error: command 'open' failed: no such file or directory
$ sudo dmesg | tail -n 11
[172324.726655] sd 0:0:0:0: [sda] Attached SCSI removable disk
[172334.342045] usb 1-1: USB disconnect, device number 98
[172334.346852] device offline error, dev sda, sector 15 op 0x1:(WRITE) flags 0x0 phys_seg 1 prio class 3
[172334.346885] Buffer I/O error on dev sda, logical block 15, lost async page write
[172334.346932] device offline error, dev sda, sector 273 op 0x1:(WRITE) flags 0x0 phys_seg 1 prio class 3
[172334.346934] Buffer I/O error on dev sda, logical block 273, lost async page write
[172334.346965] device offline error, dev sda, sector 517 op 0x1:(WRITE) flags 0x0 phys_seg 1 prio class 3
[172334.346970] Buffer I/O error on dev sda, logical block 517, lost async page write
[172334.450931] cp: attempt to access beyond end of device
                sda: rw=0, sector=0, nr_sectors = 1 limit=0
[172334.450937] FAT-fs (sda): unable to read boot sector to mark fs as dirty
```

(where the first dmesg line "Attached SCSI removable disk" corresponds to mounting the bootloader mode mass storage device.)

FWIW, I *can* use this method to upload a binary built by arduino-IDE:

```
$ python ~/zephyrproject/zephyr/scripts/build/uf2conv.py -c -f 0xada52840 -b 0xc000 -o ~/Test.uf2 /tmp/arduino/sketches/A186DE617F0051D2152DF2EDCA6EAEC3/Test.ino.hex
$ cp ~/Test.uf2 /media/ben/FTHRSNSBOOT
$ arduino-cli monitor -p /dev/ttyACM0
working...
```